util/binfmt_misc/check.go

```diff
@@ -10,30 +10,36 @@ import (
 	"path/filepath"
 )
 
-func check(bin string) error {
+func check(bin string) (string, error) {
```

Comment:
just return error. You can access stderr from `ExitError` or you can wrap the error with extra message if you like.